pallets/schemas/src/lib.rs

```diff
@@ -104,6 +109,20 @@ pub mod pallet {
 		/// Maximum number of schemas that can be registered
 		#[pallet::constant]
 		type MaxSchemaRegistrations: Get<SchemaId>;
+
+		/// The origin that is allowed to create providers
+		type CreateSchemaOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
```

Comment:
We can remove `CreateSchemaOrigin` and just stick with the `let sender = ensure_signed(origin)?;` in the function since that is all this is doing now.